src/Relationships/OneHasMany.php

```diff
@@ -15,31 +15,40 @@ class OneHasMany extends HasMany
 {
 	public function getEntitiesForPersistence()
 	{
-		$entities = [];
-		foreach ($this->toAdd as $add) {
-			$entities[] = $add;
-		}
-		foreach ($this->toRemove as $remove) {
-			if ($remove->isPersisted()) {
-				$entities[] = $remove;
-			}
-		}
 		if ($this->collection !== null || $this->wasLoaded) {
-			foreach ($this->getIterator() as $entity) {
-				$entities[] = $entity;
+			$entities = iterator_to_array($this->getIterator());
+
+		} else {
+			$entities = $this->added + $this->toAdd;
```

Comment:
a tady to presne nechapu, jak se neco bude dostat do added, alepritom nebyt wasLoaded?